presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```
@@ -0,0 +1,164 @@
+import static javax.naming.Context.SECURITY_CREDENTIALS;
+import static javax.naming.Context.SECURITY_PRINCIPAL;
+
+public class LdapFilter
+        implements Filter
+{
+    private static final Logger LOG = Logger.get(LdapFilter.class);
+    public static final String AUTHENTICATION_TYPE = "basic";
+    public static final String LDAP_CONTEXT_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+    private final String ldapBaseDn;
+    private final Hashtable<String, String> environment = new Hashtable<>();
+    private DirContext context;
+
+    @Inject
+    public LdapFilter(LdapServerConfig config)
+    {
+        ldapBaseDn = config.getBaseDn();
+
+        environment.put(INITIAL_CONTEXT_FACTORY, LDAP_CONTEXT_FACTORY);
+        environment.put(PROVIDER_URL, config.getLdapUrl());
+        try {
+            context = new InitialDirContext(environment);
+        }
+        catch (NamingException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain nextFilter)
+            throws IOException, ServletException
+    {
+        // skip auth for http
+        if (!servletRequest.isSecure()) {
+            LOG.debug("Skipping LDAP authentication in HTTP.");
+            nextFilter.doFilter(servletRequest, servletResponse);
+            return;
+        }
+
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+        String header = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+        if (header != null) {
+            String[] parts = header.split("\\s+");
+            if (parts.length == 2 && parts[0].equalsIgnoreCase(AUTHENTICATION_TYPE)) {
+                String[] credentials = B64Code.decode(parts[1], StandardCharsets.ISO_8859_1).split(":");
+                if (credentials.length == 2) {
```

You should rather do `checkState` for all the sanity validations instead of just silently ignore them and send the regular challenge.